src/Minimap.ts

```diff
@@ -0,0 +1,288 @@
+/**
+ * This is an extension adds support for adding a minimap to one of the map's control containers.
+ */
+
+import { Map } from "./Map";
+import { DOMcreate, DOMremove } from "./tools";
+
+import type {
+  ControlPosition,
+  FillLayerSpecification,
+  GeoJSONSource,
+  IControl,
+  LineLayerSpecification,
+  StyleOptions,
+  StyleSpecification,
+  StyleSwapOptions,
+} from "maplibre-gl";
+import type { MapOptions } from "./Map";
+import type { MapStyleVariant, ReferenceMapStyle } from "@maptiler/client";
+
+export interface ParentRect {
+  lineLayout: LineLayerSpecification["layout"];
+  linePaint: LineLayerSpecification["paint"];
+  fillPaint: FillLayerSpecification["paint"];
+}
+
+export interface MinimapOptionsInput {
+  /**
+   * Style of the map. Can be:
+   * - a full style URL (possibly with API key)
+   * - a shorthand with only the MapTIler style name (eg. `"streets-v2"`)
+   * - a longer form with the prefix `"maptiler://"` (eg. `"maptiler://streets-v2"`)
+   */
+  style?: ReferenceMapStyle | MapStyleVariant | StyleSpecification | string;
+
+  /**
+   * Set the zoom difference between the parent and the minimap
+   * If the parent is zoomed to 10 and the minimap is zoomed to 8, the zoomAdjust should be 2
+   * Default: -4
+   */
+  zoomAdjust?: number;
+
+  /** Set a zoom of the minimap and don't allow any future changes */
+  lockZoom?: number;
+
+  /** Adjust the pitch only if the user requests */
+  pitchAdjust?: boolean;
+
+  /** Set CSS properties of the container using object key-values */
+  containerStyle?: CSSStyleDeclaration & Record<string, string>;
+
+  /** Set the position of the minimap at either "top-left", "top-right", "bottom-left", or "bottom-right" */
+  position?: ControlPosition;
+
+  /** Set the parentRect fill and/or line options */
+  parentRect?: ParentRect;
+}
+
+export interface MinimapOptions extends MapOptions {
+  zoomAdjust: number;
+  pitchAdjust: boolean;
+  containerStyle: Record<string, string>;
+  parentRect?: ParentRect;
+}
+
+export default class Minimap implements IControl {
+  #options: MinimapOptions;
+  #minimap!: Map;
+  #parentMap!: Map;
+  #container!: HTMLElement;
+  #parentRect?: GeoJSON.Feature<GeoJSON.Polygon>;
+  #differentStyle = false;
+  #desync?: () => void;
+  constructor(options: MinimapOptionsInput, mapOptions: MapOptions) {
+    // check if the style is different
+    if (options.style !== undefined) this.#differentStyle = true;
+    // set options
+    this.#options = {
+      // set defaults
+      zoomAdjust: -4,
+      pitchAdjust: false,
+      containerStyle: {
+        border: "1px solid #000",
+        width: "500px",
+        height: "300px",
```

Comment:
From what I could see, if the container width and height are not provided in the exact same proportion of the main map, then the minimap is stretched:
<img width="326" alt="Screenshot 2023-09-22 at 11 01 40" src="https://github.com/maptiler/maptiler-sdk-js/assets/1303310/59d959da-a7f9-44e6-92f0-59e9883a92f7">